realm/realm-library/src/androidTest/java/io/realm/entities/AllJavaTypes.java

```diff
@@ -44,6 +44,8 @@
     public static final String   INVALID_LINKED_BINARY_FIELD_FOR_DISTINCT = AllJavaTypes.FIELD_OBJECT + "." + AllJavaTypes.FIELD_BINARY;
     public static final String[] INVALID_LINKED_TYPES_FIELDS_FOR_DISTINCT = new String[]{FIELD_OBJECT + "." + FIELD_BINARY, FIELD_OBJECT + "." + FIELD_OBJECT, FIELD_OBJECT + "." + FIELD_LIST};
 
+    public static final long DEFAULT_PRIMARYKEY_INDEX = 3;
```

Comment:
I'll remove the line and use `RealmObjectSchema.getPrimaryKey()`. 

At this point, I think there probably is a merit to remove all tests unrelated to migration in this PR though.